src/authhelper.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+  authhelper.cpp
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+  Copyright (C) 2016-2021 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Lieven Hey <lieven.hey@kdab.com>
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <KAuth>
+#include <QCoreApplication>
+#include <QFile>
+#include <QObject>
+#include <QProcess>
+
+class AuthHelper : public QObject
+{
+    Q_OBJECT
+public:
+    AuthHelper() = default;
+
+public slots:
+    KAuth::ActionReply elevate(const QVariantMap& args)
+    {
+        // make install installs the helper to /usr/lib/kauth/ and the elevate script to /usr/lib/libexec
+        const QString script = qApp->applicationDirPath() + QLatin1String("/../libexec/elevate_perf_privileges.sh");
+
+        if (!QFile::exists(script)) {
+            return KAuth::ActionReply::HelperErrorReply();
+        }
+
+        QProcess process;
+        process.start(script, {args[QStringLiteral("output")].toString()});
+
+        if (!process.waitForFinished(1000))
```

Comment:
remove the `!` - this whole branch logic here is flipped - did you even test this? :stuck_out_tongue:

Reply:
The registers some signal handlers that will be executed when the script is terminated. After it elevated the privileges it will wait until it receives the kill signal. Then it will reset the privileges and terminate.
So while sounding wrong this is the correct behavior that will report an error if the script terminates before 1s is passed.
The same behavior can be found in perfrecord.cpp:121

Reply:
it will report _success_ when the script terminates before 1s has passed!

and it will otherwise report an error when it finished successfully within 1s.

and if the script is supposed to be killed, we would need to launch the process in a detached manner and submit the PID as a reply from the KAuth helper, no?

Reply:
The helper waits now for a killing signal to stop the script.